contracts/serverless-v2/Executors.sol

```diff
@@ -0,0 +1,534 @@
+        _disableInitializers();
+
+        if(address(_token) == address(0))
+            revert ExecutorsZeroAddressToken();
+        if(_minStakeAmount == 0)
+            revert ExecutorsZeroMinStakeAmount();
+
+        TOKEN = _token;
+        MIN_STAKE_AMOUNT = _minStakeAmount;
+        SLASH_COMP_FOR_JOB_OWNER = _slashCompForJobOwner;
+        SLASH_COMP_FOR_GATEWAY = _slashCompForGateway;
+        SLASH_PERCENT = _slashPercent;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    function supportsInterface(
+        bytes4 interfaceId
+    )
+        public
+        view
+        virtual
+        override(ERC165Upgradeable, AccessControlUpgradeable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function _authorizeUpgrade(
+        address /*account*/
+    ) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    error ExecutorsZeroAddressAdmin();
+
+    function initialize(
+        address _admin,
+        EnclaveImage[] memory _images
+    ) public initializer {
+        if(_admin == address(0))
+            revert ExecutorsZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+        __AttestationAuther_init_unchained(_images);
+        __TreeUpgradeable_init_unchained();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable TOKEN;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable MIN_STAKE_AMOUNT;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_COMP_FOR_JOB_OWNER;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_COMP_FOR_GATEWAY;
+
+    // assuming that slash percent will be an integer in the range 0-100
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_PERCENT;
+
+    bytes32 public constant JOBS_ROLE = keccak256("JOBS_ROLE");
+
+    //-------------------------------- Executor start --------------------------------//
+
+    modifier isValidExecutor(
+        address _executor
+    ) {
+        if(executors[_executor].enclaveAddress == address(0))
+            revert ExecutorsInvalidExecutor();
+        _;
+    }
+
+    struct Executor {
+        address enclaveAddress;
+        uint256 jobCapacity;
+        uint256 activeJobs;
+        uint256 stakeAmount;
+        bool status;
```

Comment:
rename to `draining`

Reply:
updated this. 
Previously, while executor registration, status was assigned to be true. 
But now, while registration, draining would be kept as false.